tools/generate_report.rb

```diff
@@ -1,31 +1,52 @@
 #!/usr/bin/env ruby
 require File.expand_path('../config/environment', __dir__)
+require 'optimist'
 
 include ActionDispatch::Routing::UrlFor
 include Rails.application.routes.url_helpers
 
-USER_ID       = "admin".freeze
-REPORT_PARAMS = {:userid => USER_ID, :mode => "async", :report_source => "Requested by user"}.freeze
+options = Optimist.options(ARGV) do
+  banner "USAGE:  #{__FILE__} -u <name of userid which runs report, default: admin>              \n" \
+         "        #{__FILE__} -n <name of report or id of report, default: last created report>  \n" \
+         "        #{__FILE__} -l <log level, if this option is set log messages are forwarded to \n" \
+         "                                       standard output, allowed levels are debug, info, warn, error>  \n" \
 
-def report_from_args
-  if ARGV.empty?
+  opt :user, "userid", :short => "u", :type => String, :default => "admin"
```

Comment:
The type is not necessary because you provided a default.  If you want to leave it for consistency with other options, use the symbol style like you did in the other ones.